optuna/study/study.py

```diff
@@ -195,7 +197,7 @@ def direction(self) -> StudyDirection:
 
         return self.directions[0]
 
-    @property
+    @cached_property
```

Comment:
This cache logic is introduced in another pull request. So could you revert this change?

Reply:
I have reverted the change.
https://github.com/optuna/optuna/pull/4240/commits/ee429a3937986549731bf96168632b1286b65f6b